text/0000-user-ids.md

```diff
@@ -0,0 +1,42 @@
+# Secure runtime environments
+
+## Summary
+
+This RFC recommends that Paketo Buildpacks should in general strive to avoid making run-time modifications to the Buildpacks layers or the application directory by default. Instead, they should consider the output image to be read-only by default and explicitly set the parts that need to be writeable during runtime.
+
+
+## Motivation
+
+As of Buildpacks [RFC 0085](https://github.com/buildpacks/rfcs/blob/main/text/0085-run-uid.md) the Cloud Native Buildpacks project recommends that the run-time and build-time user ids be different. The motivation behind this RFC was to improve the security of output images and prevent arbitrary runtime modifications to built artifacts.
+
+The run-time and build-time user can potentially share the same group id if they wish to make modifications to the output image during run-time, by changing the permissions of the files they wish to modify during run-time to be group writable during the build process. Ideally most buildpacks should use the temporary directories or other known scratch directories for ephemeral runtime files.
+
+## Detailed Explanation
+
+This RFC recommends that we identify buildpacks that making runtime modifications to the application directory or buildpack layers and modify them to instead use a temporary directory - or have them make these files group writable instead. All the current Paketo stacks' build-time and run-time users belong to a unique group so changing the file permissions to be group writable should have no additional security implications for users using the Paketo stacks.
+
+
+## Rationale and Alternatives
+
+
+The rationale behind this RFC is explained in the upstream RFC - [RFC 0085](https://github.com/buildpacks/rfcs/blob/main/text/0085-run-uid.md)
+
+In the future we could also modify the Paketo stacks to have different run-time/build-time user ids once all the buildpacks follow this practice for a more secure experience.
+
+## Implementation
+
+At a high level this would involve the following - 
+
+- Use temporary directories where possible to write files
+- Do the equivalent of `chmod g+w` for files that we are expecting to modify during runtime.
+
```

Comment:
Similar to #136, I think it would be helpful to define a more concrete plan here.

For example:

1. Create an issue to review & catalog our present compliance.
2. Open issues for non-compliant buildpacks to investigate making the change.
3. Open issues to switch where possible.
4. When buildpacks are tested, switch stacks to use different run/build ids.
5. Update Paketo documentation to explain this security precaution and any steps a user might need to take.

Reply:
The easiest and most comprehensive way to check for compliance might be to create a test stack that has different build and run user IDs. Integration and smoke tests across the repos would probably point out most of the places that will require changes.

Reply:
+1 to the idea of creating a more concrete plan

Also, It might be good to include some examples in this RFC including common patterns (i.e. ephemeral config file, log file, etc)